daemon/list.go

```diff
@@ -179,49 +211,8 @@ func (daemon *Daemon) filterByNameIDMatches(view *container.View, filter *listCo
 	return cntrs, nil
 }
 
-// reduceContainers parses the user's filtering options and generates the list of containers to return based on a reducer.
-func (daemon *Daemon) reduceContainers(ctx context.Context, config *types.ContainerListOptions, reducer containerReducer) ([]*types.Container, error) {
-	if err := config.Filters.Validate(acceptedPsFilterTags); err != nil {
-		return nil, err
-	}
-
-	var (
-		view       = daemon.containersReplica.Snapshot()
-		containers = []*types.Container{}
-	)
-
-	filter, err := daemon.foldFilter(ctx, view, config)
-	if err != nil {
-		return nil, err
-	}
-
-	// fastpath to only look at a subset of containers if specific name
-	// or ID matches were provided by the user--otherwise we potentially
-	// end up querying many more containers than intended
-	containerList, err := daemon.filterByNameIDMatches(view, filter)
-	if err != nil {
-		return nil, err
-	}
-
-	for i := range containerList {
-		t, err := daemon.reducePsContainer(ctx, &containerList[i], filter, reducer)
-		if err != nil {
-			if err != errStopIteration {
-				return nil, err
-			}
-			break
-		}
-		if t != nil {
-			containers = append(containers, t)
-			filter.idx++
-		}
-	}
-
-	return containers, nil
-}
-
-// reducePsContainer is the basic representation for a container as expected by the ps command.
-func (daemon *Daemon) reducePsContainer(ctx context.Context, container *container.Snapshot, filter *listContext, reducer containerReducer) (*types.Container, error) {
+// formatContainerForPs formats a container for the ps output.
+func (daemon *Daemon) formatContainerForPs(ctx context.Context, container *container.Snapshot, filter *listContext) (*types.Container, error) {
```

Comment:
I'm thinking of a good name for this function. `format` sounds like something that wouldn't mutate anything.
We could move the `daemon.refreshImage` call to avoid that, but then we wouldn't take result of `includeContainerInList` into account..

I think we can go one step further and completely inline `formatContainerForPs` in `Containers`. We could also get rid of `errStopIteration`. And we won't have to come up with a name 🤣.

Reply:
Hey, this branch has become a mess since this is my first contribution and I made some wrong things with git. Can I close this and open a new PR?

Reply:
Sure 👍